[docs] Direct users from Material UI to MUI Base for duplicated components (#35293)
mui/material-ui: docs/data/material/components/click-away-listener/click-away-listener.md

# Click-Away Listener

<p class="description">The Click-Away Listener component detects when a click event happens outside of its child element.</p>

## MUI Base

:::warning
Click-Away Listener is a part of the standalone [MUI Base](/base/getting-started/overview/) component library.

> It is currently re-exported from `@mui/material` for your convenience, but will may be removed from this package in a future major version, after `@mui/base` gets a stable release.

Reviewer: ```suggestion
It is currently re-exported from `@mui/material` for your convenience, but will be removed from this package in a future major version, after `@mui/base` gets a stable release.
```

# Click-Away Listener

<p class="description">The Click-Away Listener component detects when a click event happens outside of its child element.</p>

## Migrate to MUI Base

> :::warning

Reviewer 2: The info is safe, so maybe?

```suggestion
:::info
```

Click-Away Listener is now part of the standalone [MUI Base](/base/getting-started/overview/) component library.
It is currently re-exported in `@mui/material` for convenience, but it may be deprecated in a future major version of Material UI.

> To continue using this component and avoid future breaking changes, we strongly recommend that you [install MUI Base](/base/getting-started/installation/) and import this component from the `@mui/base` package:

Reviewer: I wouldn't put this recommendation just yet. MUI Base is still in alpha, and many organizations are reluctant to install unstable packages (even though what we reexport from Material UI is the same thing). I'd wait until beta to officially recommend this. As for now, we can point readers to the Base docs and say they can import from either package. I suppose we can mention that we plan to move it to Base completely at some point).

> Click-Away Listener is now part of the standalone [MUI Base](/base/getting-started/overview/) component library.
> It is currently re-exported in `@mui/material` for convenience, but it may be deprecated in a future major version of Material UI.
> 
> To continue using this component and avoid future breaking changes, we strongly recommend that you [install MUI Base](/base/getting-started/installation/) and import this component from the `@mui/base` package:
> 
> ```js
> import ClickAwayListener from '@mui/base/ClickAwayListener';
> ```
> 
> Please refer to the [Click-Away Listener](/base/react-click-away-listener/) component page in the MUI Base docs for examples and details on usage.

Author: @michaldudak how about this text instead?

```suggestion
Click-Away Listener is a part of the standalone [MUI Base](/base/getting-started/overview/) component library.
It is currently re-exported in `@mui/material` for your convenience, but it may be removed from this package in a future major version, after `@mui/base` gets a stable release.

Please refer to the [Click-Away Listener](/base/react-click-away-listener/) component page in the MUI Base docs for examples and details on usage.
```

Reviewer: I like it!
One remark: in the second paragraph, shouldn't it say "It is currently re-exported ~in~ **from** `@mui/material`..."?

Author: Good question! I couldn't decide between the two here, but re-reading it now I think you're right that it's better that way.

Reviewer 2: I think that we can commit to it, it's pretty much guaranteed:

```diff
-It is currently re-exported in `@mui/material` for your convenience, but it may be removed from this package in a future major version, after `@mui/base` gets a stable release.
-It is currently re-exported in `@mui/material` for your convenience, but it will be removed from this package in a future major version, after `@mui/base` gets a stable release.
```
